components/error-toast.tsx

```
@@ -0,0 +1,286 @@
+      setToasts((prev) =>
+        prev.map((toast) => {
+          const timerId = timers.get(toast.id)
+          return timerId ? { ...toast, timerId } : toast
+        })
+      )
+    }
+
+    return () => {
+      timers.forEach((timer) => clearTimeout(timer))
+    }
+  }, [toasts, handleDismiss])
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)